### ISSUE TITLE
when describing from table, multi labels is broken

### ISSUE THREAD

**Issue by Yann-CV** [goldener-data/goldener#202]

**Comment by AshleyAHuang** [comment]
Confirmed reproducible on current main — but it's broader than "from
table" suggests: it happens with plain Dataset input too, whenever
`collate_fn` is left at its default (None).

## Smallest possible repro (isolates the actual root cause)

```python
from torch.utils.data.dataloader import default_collate

batch = [
    {"idx": 0, "label": ["class_1", "class_2"]},
    {"idx": 1, "label": ["class_1", "class_2"]},
]
print(default_collate(batch)["label"])
# [('class_1', 'class_1'), ('class_2', 'class_2')]
# Expected (per-sample lists preserved): [['class_1', 'class_2'], ['class_1', 'class_2']]
```

PyTorch's `default_collate` transposes list-valued dict fields across
the batch dimension by position, instead of preserving one list per
sample. `transform_batch_from_multilabel_to_independent_labels`
(`goldener/utils.py:465`) then does `enumerate(batch[label_key])`
expecting each entry to be one sample's full label list — but with the
transposed structure, each "sample" it sees is actually one label
*position* collected across every sample in the batch.

## End-to-end repro against `GoldDescriptor.describe_in_table`

Two samples, both labeled `["class_1", "class_2"]`, `collate_fn=None`:

```python
desc = GoldDescriptor(
    table_path="unit_test.repro",
    embedder=embedder,
    vectorizer=vectorizer,
    label_key="label",
    to_keep_schema={"label": pxt.String},
    batch_size=2,
    collate_fn=None,
    allow_existing=False,
)
description = desc.describe_in_table(DummyDataset([
    {"idx": 0, "data": torch.zeros(3, 4, 4), "label": ["class_1", "class_2"]},
    {"idx": 1, "data": torch.zeros(3, 4, 4), "label": ["class_1", "class_2"]},
]))
```

**Expected:** each sample split into two rows, one per label (both
samples get both `class_1` and `class_2` rows).
**Actual:** idx 0 only ever gets `class_1`, idx 1 only ever gets
`class_2` — each sample silently loses one of its two labels.

Confirmed this does NOT happen when an explicit `collate_fn` is
provided that preserves per-sample list structure (e.g. the pattern
used in `test_describe_in_dataset_from_table_with_vectorizer_and_multitarget`)
— that's likely why the existing multi-target tests didn't catch this.

Not proposing a fix here, just confirming + isolating the root cause
so whoever picks this up doesn't have to re-derive it.

**Comment by Yann-CV** [comment]
@AshleyAHuang for info this is happening because we are not really treating labels the default way for Pytorch. In the multilabel situation, the model is usually receiving multi hot encoded tensors expressing which class is present and not strings.

I am creating an issue to see how we could do something similar here.

**Comment by Yann-CV** [comment]
@AshleyAHuang I allowed myself to update the description from your findings, thanks again!

Do you still want to work on this? If yes, can you please assign yourself to it?

**Comment by AshleyAHuang** [comment]
yep, still on it! tried to self-assign but github won't let me (need triage perms on the repo) — mind assigning me? fix incoming on #250 in the meantime

**Comment by Yann-CV** [comment]
done ! I will look in repo settings to see if I can fix this

**Comment by Yann-CV** [comment]
@AshleyAHuang I gave you the triage right! You should be able to assign yourself now ;-)

**Comment by Yann-CV** [comment]
done in #250
